tests/custom-elements/reactions/HTMLButtonElement.html

```diff
@@ -0,0 +1,112 @@
+    testReAtWithPNodeWithCoValues(jsAtName, coAtName, jsAtValue1, jsAtValue1, jsAtValue2, jsAtValue2, name, elementName, pElementName, interfaceName);
+}
+
+//In parent node, sub node's observeAttribute which depends another attribute can enqueue by changing attribute value
+//Re->reflect, At->attribute, De->depdent, Co->content
+function testReAtWithDeAtWithCoValues(jsAtName, coAtName, deAtName, jsAtValue1, coAtValue1, jsAtValue2, coAtValue2, deAtValue, name, elementName, pElementName, interfaceName) {
+    var parentElement = document.createElement(pElementName);
+    document.body.appendChild(parentElement);
+
+    test(() => {
+        var element = define_build_in_custom_element([coAtName], interfaceName, elementName);
+        var instance = document.createElement(elementName, { is: element.name });
+
+        assert_array_equals(element.takeLog().types(), ['constructed']);
+        parentElement.appendChild(instance);
+        assert_array_equals(element.takeLog().types(), ['connected']);
+        instance.setAttribute(deAtName, deAtValue);
+        instance[jsAtName] = jsAtValue1;
+        var logEntries = element.takeLog();
+        assert_array_equals(logEntries.types(), ['attributeChanged']);
+        assert_attribute_log_entry(logEntries.last(), {name: coAtName, oldValue: null, newValue: coAtValue1, namespace: null});
+
+    }, name + ' must enqueue an attributeChanged reaction when adding a new attribute');
+
+    test(() => {
+        var element = define_build_in_custom_element([coAtName], interfaceName, elementName);
+        var instance = document.createElement(elementName, { is: element.name});
+        parentElement.appendChild(instance);
+        instance.setAttribute(deAtName, deAtValue);
+        instance[jsAtName] = jsAtValue1;
+
+        assert_array_equals(element.takeLog().types(), ['constructed', 'connected', 'attributeChanged']);
+        instance[jsAtName] = jsAtValue2;
+        var logEntries = element.takeLog();
+        assert_array_equals(logEntries.types(), ['attributeChanged']);
+        assert_attribute_log_entry(logEntries.last(), {name: coAtName, oldValue: coAtValue1, newValue: coAtValue2, namespace: null});
+
+    }, name + ' must enqueue an attributeChanged reaction when replacing an existing attribute');
+
+    parentElement.parentNode.removeChild(parentElement);
+}
+
+//Package JS attribute depends another atrribute
+function testReAtWithDeAt(jsAtName, coAtName, deAtName, jsAtValue1, jsAtValue2, deAtValue, name, elementName, pElementName, interfaceName) {
+    testReAtWithDeAtWithCoValues(jsAtName, coAtName, deAtName, jsAtValue1, jsAtValue1, jsAtValue2, jsAtValue2, deAtValue, name, elementName, pElementName, interfaceName);
+}
+
+testReflectAttributeWithContentValues('autofocus', 'autofocus', true, 'true', false, 'false', 'autofocus on HTMLButtonElement', 'button', HTMLButtonElement);
```

Comment:
move following test function calls to the beginning of <script>

Reply:
Move function calls to the beginning of <script>